docs/user/openstack/install_upi.md

```diff
@@ -0,0 +1,861 @@
+- Nova flavors
+  - This document will use `m1.xlarge` for masters and `m1.large` for workers
+- An external subnet you want to use for floating IP addresses
+  - This document will use `external`
+- The `openshift-install` binary in your `$PATH`
+- The [Red Hat CoreOS][rhcos] image in Glance
+  - This document will use the `rhcos` image name
+- A subnet range for the Nova servers / OpenShift Nodes
+  - This range must not conflict with your existing network
+  - This document will use `192.0.2.0/24`
+- A cluster name you will want to use
+  - This document will use `openshift`
+- A base domain
+  - This document will use `example.com`
+- OpenShift Pull Secret
+- DNS infrastructure you can configure (you'll need to add two records there)
+
+[rhcos]: https://www.openshift.com/learn/coreos/
+
+The OpenShift API URL will be generated from the cluster name and base domain. E.g.: `https://api.openshift.example.com:6443/`
+
+
+You can validate most of the above by running the following commands:
+
+```sh
+$ export OS_CLOUD=openstack
+$ openstack image show rhcos
+$ openstack network show external
+$ openstack flavor show m1.xlarge
+$ openstack flavor show m1.large
+$ which openshift-install
+```
+
+They should all succeed.
+
+
+## OpenShift Configuration Directory
+
+All the configuration files, logs and installation state is kept in a single directory:
+
+```sh
+$ mkdir -p openstack-upi
+$ cd openstack-upi
+```
+
+## API and Ingress Floating IP Addresses
+
+```sh
+$ openstack floating ip create --description "OpenShift API" <external>
+=> 203.0.113.23
+$ openstack floating ip create --description "OpenShift Ingress" <external>
+=> 203.0.113.19
+```
+
+**NOTE**: throughout this document, we will use `203.0.113.23` as the public IP address for the OpenShift API endpoint and `203.0.113.19` as the public IP for the ingress (`*.apps`) endpoint.
+
+## Install Config
+
+Run the `create install-config` subcommand and fill in the desired entries:
+
+```sh
+$ openshift-install create install-config
+? SSH Public Key </home/user/.ssh/id_rsa>
+? Platform <openstack>
+? Cloud <openstack>
+? ExternalNetwork <external>
+? APIFloatingIPAddress <203.0.113.23>
+? FlavorName <m1.xlarge>
+? Base Domain <example.com>
+? Cluster Name <openshift>
+```
+
+Most of these are self-explanatoray. `Cloud` is the cloud name in your `clouds.yaml` i.e. what's set as your `OS_CLOUD` environment variable.
+
+Afterwards, you should have `install-config.yaml` in your current directory:
+
+```sh
+$ tree
+.
+└── install-config.yaml
+```
+
+### Fix the Node Subnet
+
+The installer added a default IP range for the OpenShift nodes. It must match the range for the Neutron subnet you'll create later on.
+
+We're going to use a custom subnet to illustrate how that can be done.
+
+
+Our range will be `192.0.2.0/24` so we need to add that value to
+`install-config.yaml`. Look under `networking` -> `machineCIDR`.
+
+This command will do it for you:
```

Comment:
General question, please just ignore if already discussed or out of scope.

I read that [`install-config.yaml` is part of the stable Installer API](https://github.com/openshift/installer/blob/master/docs/user/overview.md#multiple-invocations). Wouldn't it be more straightforward to provide here an `install-config.yaml` template where to fill IPs and CIDRs and flavours by hand, rather than asking the users to feed `openshift-install` interactively and _then_ change the output YAML?

Personally, as soon as I figured out that I could fine-tune my cluster by editing `install-config.yaml`, I have never come back again to `create cluster` without providing a YAML.

Reply:
Possibly, but it's what AWS UPI does as well. I didn't want to strike on our own on things like that: https://github.com/openshift/installer/blob/master/docs/user/aws/install_upi.md#create-configuration